src/EscapeAnalysis.jl

```diff
@@ -136,13 +136,42 @@ end
 # analysis
 # ========
 
+# FIXME alias analysis
+#    Especially, we should take care of aliasing of call argument and return value
+#    ```julia
+#    @noinline function f_return_escape(a)
+#        println("hi") # prevent inlining
+#        return a
+#    end
+#    result = @eval m $analyze_escapes() do
+#        obj = Ref("foo")           # should be "return escape"
+#        ret = f_return_escape(obj)
+#        return ret                 # alias of `obj`
+#    end
+#
+# TODO flow-sensitivity
+#    Main motivation here is finalizer elision, e.g.:
+#    ```julia
+#    let
+#        m = make_m()
+#        if cond
+#            return nothing # <= insert a finalizer call here
+#        end
+#        return m
+#    end
+#    ```
+#    We can do setup separate states for each statement (probably better with sparse states
+#    since there can be many statements), or encoding the quivalent information into the lattice
+
```

Comment:
@vtjnash I'd like to hear your ideas on how to address these fixes/enhancements.

As for the finalizer elision, I guess we can:
- turn `ReturnEscape` into `Vector{Int}`
- and each return site, insert early `finalize` calls if the statement isn't neither of `ReturnEscape`, `GlobalEscape` (and we just ignore `ThrownEscape`)

Reply:
Ops, I meant these lines: https://github.com/aviatesk/EscapeAnalysis.jl/blob/f03342595148ffb45d599c97a19e394de2c899c6/src/EscapeAnalysis.jl#L139-L164

Reply:
The first point is addressed thanks to the idea of `from_interprocedural`: https://github.com/aviatesk/EscapeAnalysis.jl/pull/36#discussion_r684347301